### PR TITLE
Change to f-strings

### PR DESCRIPTION
**Pull request by mattlorimor**
This changes all string replacements to be done via the more-contemporary [f-string](https://realpython.com/python-f-strings/) way.

### Things to double check
- [ ] each f-string starts with an `f`
- [ ] nested f-strings function as expected (i think there was only one)
- [ ] the multi-line f-string doing the header build [works as expected](https://realpython.com/python-f-strings/#multiline-f-strings)

I'll call out in the PR diff places to triple check.

### For Site Improvements

- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by the Bug Alert team if needed.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have run Pelican locally to ensure my changes have not disrupted the look, feel, or functionality of the site.
- [x] I have checked my code and corrected any misspellings.
- [x] I have added a pull request description which details the nature of this change.

### CODE REVIEW

File: notice_communication/__main__.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ def main():
 
     if os.getenv('TWITTER_BEARER_TOKEN'):
         twitter = get_twitter_client()
-        url = "https://bugalert.org/%s" % filename.replace('md', 'html')
-        tweet = "%s %s #BugAlertNotice" % (("%s..." % summary[:220] if len(summary) > 220 else summary), url)
+        url = f"https://bugalert.org/{filename.replace('md', 'html')}"
+        tweet = f"{(f'{summary[:220] if len(summary) > 220 else summary)}...'} {url} #BugAlertNotice"
```
Comment (mattlorimor) [inline notice_communication/__main__.py]: nested f-string worth a triple check

File: subscription_management/lambda_function.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 def respond_error(err, origin, status=400):
     return {
         'statusCode': status,
-        'body': "{\"error\": \"%s\"}" % err,
+        'body': f"{{\"error\": \"{err}\"}}",
```
Comment (mattlorimor) [inline subscription_management/lambda_function.py:35]: `{{` gets a literal `{` in there; double check that it works

File: subscription_management/lambda_function.py [hunk 3]
```diff
@@ -255,27 +262,27 @@ def upload_telephony_contact_list(category):
...
+            phone_file.write(f"Bugs,Allert,bugs.allert@example.com,1{contact.get('phone_number')}\n")
 
     sms_file.close()
     phone_file.close()
 
     # Now put them on telephony services
     sms_file_id = upload_contacts('/tmp/sms.csv')
     phone_file_id = upload_contacts('/tmp/phone.csv')
-    print("sms_file_id: %s" % sms_file_id)
-    print("phone_file_id: %s" % sms_file_id)
```
Comment (mattlorimor) [inline subscription_management/lambda_function.py:269]: I assume `sms_file_id` being here twice was an accident. Fixed it.

**Comment by mattlorimor** [comment]
Merged in main and resolved the conflicts.

I also cleaned up some explicit `open` and `close` operations and changed them to `with` statements. Double check those to make sure I didn't break something expecting it to be done that way.

File: notice_communication/__main__.py (hunk 1 above)
Comment (sullivanmatt) [reply notice_communication/__main__.py]:
```
>>> url = "https://bugalert.org/content/notice/2022-01-02-test.md"
>>> summary = "A remote code execution vulnerability has been found in the popular Java logging library Log4j. This issue is easily exploited in common configurations, and has been assigned a bug alert severity of 'critical'."
>>> tweet = f"{(f'{summary[:220] if len(summary) > 220 else summary)}...'} {url} #BugAlertNotice"
  File "<stdin>", line 1
    tweet = f"{(f'{summary[:220] if len(summary) > 220 else summary)}...'} {url} #BugAlertNotice"
                                                                                                 ^
SyntaxError: f-string: closing parenthesis '}' does not match opening parenthesis '('
```

File: subscription_management/lambda_function.py (hunk 2 above)
Comment (sullivanmatt) [reply subscription_management/lambda_function.py:35]: Looks good. 

```
>>> err = "Lorem ipsum"
>>> f"{{\"error\": \"{err}\"}}"
'{"error": "Lorem ipsum"}'

```

File: subscription_management/lambda_function.py [hunk 4]
```diff
@@ -62,7 +62,14 @@ def respond_success(msg, origin):
...
     headers = {k.lower():v for k,v in headers.items()}
-    logger.info("%s %s %s %s %s %s" % (headers.get('x-forwarded-for', 'no-ip'), event.get('httpMethod', 'no-method'), event.get('path', 'no-path'), headers.get('user-agent', 'no-ua'), headers.get('referer', 'no-referer'), headers.get('origin', 'no-origin')))
+    logger.info(
+        f"{headers.get('x-forwarded-for', 'no-ip')} "
+        f"{event.get('httpMethod', 'no-method')} "
+        f"{event.get('path', 'no-path')} "
+        f"{headers.get('user-agent', 'no-ua')} "
+        f"{headers.get('referer', 'no-referer')} "
+        f"{headers.get('origin', 'no-origin')}"
+    )
```
Comment (sullivanmatt) [inline subscription_management/lambda_function.py:72]: Thanks for cleaning this up, it was badly needed.

File: subscription_management/lambda_function.py [hunk 5]
```diff
@@ -101,14 +108,14 @@ def lambda_handler(event, context):
         # A future improvement could be made to authenticate this, but for the moment,
         # I don't forsee anyone DoS-ing this endpoint causing any harm.
         sms_file_id, phone_file_id = upload_telephony_contact_list(body.get('category'))
-        return respond_success("{\"status\": \"success\", \"sms_file_id\": \"%s\", \"phone_file_id\": \"%s\"}" % (sms_file_id, phone_file_id), origin)
+        return respond_success(f"{{\"status\": \"success\", \"sms_file_id\": \"{sms_file_id}\", \"phone_file_id\": \"{phone_file_id}\"}}", origin)
```
Comment (sullivanmatt) [inline subscription_management/lambda_function.py]: These lines (both before and after your changes) are really busy. Mind using your branch and PR as an opportunity to split this these into two ops, where one is construction of the json and the second is the call to `respond_success`?

File: subscription_management/lambda_function.py [hunk 6]
```diff
@@ -101,14 +108,14 @@ def lambda_handler(event, context):
...
     elif method == 'POST' and path == 'verify':
         # Fire off email to confirm the user is allowed to make subscription changes.
         locally_computed_hmac = hmac.new(hmacsecret, codecs.encode(email), hashlib.sha256).digest()
         signature = base64.urlsafe_b64encode(locally_computed_hmac).decode('utf8').rstrip("=")
-        msg_body = "Please visit the following URL to verify your email address: " \
-                   "https://bugalert.org/content/pages/my-subscriptions.html" \
-                   "?signature={}&email={}".format(signature, email)
+        msg_body = f"Please visit the following URL to verify your email address: "
+                   f"https://bugalert.org/content/pages/my-subscriptions.html"
+                   f"?signature={signature}&email={email}"
```
Comment (sullivanmatt) [inline subscription_management/lambda_function.py]:
```suggestion
        msg_body = f"Please visit the following URL to verify your email address: " \
                   f"https://bugalert.org/content/pages/my-subscriptions.html" \
                   f"?signature={signature}&email={email}"
```

Multiline requires the EOL backslashes.

File: subscription_management/lambda_function.py [hunk 7]
```diff
@@ -247,35 +254,31 @@ def upload_telephony_contact_list(category):
...
-    sms_file.close()
-    phone_file.close()
-
-    # Now put them on telephony services
-    sms_file_id = upload_contacts('/tmp/sms.csv')
-    phone_file_id = upload_contacts('/tmp/phone.csv')
-    print("sms_file_id: %s" % sms_file_id)
-    print("phone_file_id: %s" % sms_file_id)
-    return sms_file_id, phone_file_id
+    with open('/tmp/sms.csv', 'w') as sms_file, open('/tmp/phone.csv', 'w') as phone_file:
```
Comment (sullivanmatt) [inline subscription_management/lambda_function.py:258]: TIL. Had no idea you could do multiple `with` declarations.

**Review by sullivanmatt — CHANGES_REQUESTED** [review]

File: subscription_management/lambda_function.py (hunk 5 above)
Comment (mattlorimor) [reply subscription_management/lambda_function.py]: Done

**Review by sullivanmatt — APPROVED** [review]

**Pull request merged**